src/System.Drawing.Common/src/System/Drawing/Pen.Unix.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing.Drawing2D;
+using System.Runtime.InteropServices;
+
+namespace System.Drawing
+{
+    public partial class Pen
+    {
+        // libgdiplus does not implement GdipGetPenCustomEndCap, so we cache the last-known value here.
+        // Note that this value is not necessarily in sync with the true native value of this property,
+        // as it could have been set outside of the CustomEndCap property on this type.
+        private CustomLineCap _cachedEndCap;
+
+        /// <summary>
+        /// Gets or sets a custom cap style to use at the beginning of lines drawn with this <see cref='Pen'/>.
+        /// </summary>
+        public CustomLineCap CustomStartCap
+        {
+            get
+            {
+                IntPtr lineCap = IntPtr.Zero;
+                int status = SafeNativeMethods.Gdip.GdipGetPenCustomStartCap(new HandleRef(this, NativePen), out lineCap);
+                SafeNativeMethods.Gdip.CheckStatus(status);
+                if (lineCap == IntPtr.Zero)
```

Comment:
Right: libgdiplus just returns a null handle if the Pen doesn't have an associated CustomStartCap. On Windows, the accessor throws an ArgumentException if you haven't set the CustomStartCap yet.